docs/dev/contributing.md

```diff
@@ -89,11 +89,27 @@ If your commit is only shipping documentation changes or example files, and is a
 complete no-op for the test suite, please add **[skip ci]** in the commit message
 body to skip the build and give that slot to someone else who does need it.
 
-### Squash your commits
+### Pull request workflow
 
-Please rebase your changes on `main` and squash your commits whenever possible,
-it keeps history cleaner and it's easier to revert things. It also makes developers
-happier!
+The goals ordered by priority are:
+- Make PR reviews (both initial and follow-up reviews) easy for reviewers using GitHub
+- On the `main` branch, have a meaningful commit history that allows understanding (even years later) what each commit does, and why. 
```

Comment:
The lines in the patch should probably wrap at the same char limit as the rest of the file

Reply:
Not sure to understand how we have a meaningful commit history as we use squash merged (which destroys the history and therefore merges a big PR into one single commit)